### ISSUE TITLE
Crash when marking Text

### ISSUE THREAD

**Issue by felixdivo** [JetBrains/compose-jb#1247]
I got this crash on version `1.0.0-alpha4-build398` when marking `Text(...)` wrapped inside a `SelectionContainer(...)` with a mouse cursor. It caused the app to freeze forever. It is not really reproducible.

```
Exception in thread "AWT-EventQueue-0" java.lang.IllegalArgumentException: Rect::makeLTRB expected l <= r, got NaN > NaN
	at org.jetbrains.skia.Rect$Companion.makeLTRB(Rect.kt:89)
	at org.jetbrains.skia.paragraph.TextBox.<init>(TextBox.kt:10)
	at org.jetbrains.skia.paragraph.ParagraphKt._nGetRectsForRange(Native Method)
	at org.jetbrains.skia.paragraph.ParagraphKt.access$_nGetRectsForRange(Paragraph.kt:1)
	at org.jetbrains.skia.paragraph.Paragraph.getRectsForRange(Paragraph.kt:111)
	at androidx.compose.ui.text.platform.DesktopParagraph.getBoxForwardByOffset(DesktopParagraph.desktop.kt:310)
	at androidx.compose.ui.text.platform.DesktopParagraph.getBidiRunDirection(DesktopParagraph.desktop.kt:346)
	at androidx.compose.ui.text.MultiParagraph.getBidiRunDirection(MultiParagraph.kt:418)
	at androidx.compose.ui.text.TextLayoutResult.getBidiRunDirection(TextLayoutResult.kt:353)
	at androidx.compose.foundation.text.selection.MultiWidgetSelectionDelegateKt.getAssembledSelectionInfo-vJH6DeI(MultiWidgetSelectionDelegate.kt:238)
	at androidx.compose.foundation.text.selection.MultiWidgetSelectionDelegateKt.getTextSelectionInfo-yM0VcXU(MultiWidgetSelectionDelegate.kt:177)
	at androidx.compose.foundation.text.selection.MultiWidgetSelectionDelegate.updateSelection-qCDeeow(MultiWidgetSelectionDelegate.kt:60)
	at androidx.compose.foundation.text.selection.SelectionManager.updateSelection-3R_-tFg$foundation(SelectionManager.kt:702)
	at androidx.compose.foundation.text.selection.SelectionManager.startSelection-9KIMszo(SelectionManager.kt:611)
	at androidx.compose.foundation.text.selection.SelectionManager.access$startSelection-9KIMszo(SelectionManager.kt:58)
	at androidx.compose.foundation.text.selection.SelectionManager$2.invoke-d-4ec7I(SelectionManager.kt:202)
	at androidx.compose.foundation.text.selection.SelectionManager$2.invoke(SelectionManager.kt:194)
	at androidx.compose.foundation.text.selection.SelectionRegistrarImpl.notifySelectionUpdateStart-d-4ec7I(SelectionRegistrarImpl.kt:177)
	at androidx.compose.foundation.text.TextController$mouseSelectionObserver$1.onStart-3MmeM6k(CoreText.kt:522)
	at androidx.compose.foundation.text.selection.TextSelectionMouseDetectorKt$mouseSelectionDetector$2$1.invokeSuspend(TextSelectionMouseDetector.kt:112)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTaskKt.resume(DispatchedTask.kt:178)
	at kotlinx.coroutines.DispatchedTaskKt.dispatch(DispatchedTask.kt:166)
	at kotlinx.coroutines.CancellableContinuationImpl.dispatchResume(CancellableContinuationImpl.kt:397)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl(CancellableContinuationImpl.kt:431)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeImpl$default(CancellableContinuationImpl.kt:420)
	at kotlinx.coroutines.CancellableContinuationImpl.resumeWith(CancellableContinuationImpl.kt:328)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter$PointerEventHandlerCoroutine.offerPointerEvent(SuspendingPointerInputFilter.kt:513)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter.dispatchPointerEvent(SuspendingPointerInputFilter.kt:407)
	at androidx.compose.ui.input.pointer.SuspendingPointerInputFilter.onPointerEvent-H0pRuoY(SuspendingPointerInputFilter.kt:420)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:284)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:271)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:271)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:271)
	at androidx.compose.ui.input.pointer.Node.dispatchMainEventPass(HitPathTracker.kt:271)
	at androidx.compose.ui.input.pointer.NodeParent.dispatchMainEventPass(HitPathTracker.kt:151)
	at androidx.compose.ui.input.pointer.HitPathTracker.dispatchChanges(HitPathTracker.kt:88)
	at androidx.compose.ui.input.pointer.PointerInputEventProcessor.process-BIzXfog(PointerInputEventProcessor.kt:80)
	at androidx.compose.ui.platform.DesktopOwner.processPointerInput-8iAsVTc$ui(DesktopOwner.desktop.kt:340)
	at androidx.compose.ui.platform.DesktopOwners.onMousePressed(DesktopOwners.desktop.kt:419)
	at androidx.compose.ui.platform.DesktopOwners.sendPointerEvent-DHzpxOM(DesktopOwners.desktop.kt:368)
	at androidx.compose.ui.awt.ComposeLayer_desktopKt.onMouseEvent(ComposeLayer.desktop.kt:260)
	at androidx.compose.ui.awt.ComposeLayer_desktopKt.access$onMouseEvent(ComposeLayer.desktop.kt:1)
	at androidx.compose.ui.awt.ComposeLayer$3$mousePressed$1.invoke(ComposeLayer.desktop.kt:163)
	at androidx.compose.ui.awt.ComposeLayer$3$mousePressed$1.invoke(ComposeLayer.desktop.kt:162)
	at androidx.compose.ui.awt.AWTDebounceEventQueue$job$1.invokeSuspend(AWTDebounceEventQueue.desktop.kt:59)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by igordmn** [comment]
No, it is a new issue

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-rc6

**Comment by felixdivo** [comment]
Thanks @igordmn!
